Doc/library/functions.rst

```diff
@@ -561,6 +557,29 @@ are always available.  They are listed here in alphabetical order.
       Raises an :ref:`auditing event <auditing>` ``exec`` with the code object
       as the argument. Code compilation events may also be raised.
 
+   .. note::
+
+      Dynamic evaluation at run-time is not equivalent to embedding the
+      expression at the same place in a Python program and having it compiled
+      as a part of the whole program; see :ref:`dynamic-features` for details.
+
+      In particular, :func:`eval` does not have access to the :term:`nested
+      scopes <nested scope>` (non-locals) in the enclosing environment.  Of
+      note are expressions such as :term:`lambdas <lambda>`,
```

Comment:
I'm not exactly following this section, as I thought nonlocals were affected by functions and not lambdas and the others, but I copied it over here nonetheless for some comments.

Reply:
lambda is just an unnamed function, and comprehensions are also scoped, see e.g. https://portingguide.readthedocs.io/en/latest/comprehensions.html#leaking-of-the-iteration-variable

Reply:
I see I see, thanks for the link.